app/index.coffee

```diff
@@ -105,6 +107,11 @@ Api.proxy.el().one 'load', ->
       app.topBar.loginForm.signInButton.click()
       app.topBar.loginDialog.reattach()
 
+    $(window).bind('beforeunload', (e) ->
+        AnalyticsLogger.logEvent User.current?.zooniverse_id,Subject.current?.zooniverseId,'leave',''
+        event.preventDefault()
```

Comment:
Ah. I think `beforeunload` only shows a dialog if you return something other than `undefined`. I don't entirely know how this event works though, so I'd check some docs and make sure.